src/Ninja/G1CP/Content/Fixes/Gamesave/fix224_OrcPriestFallDamage.d

```diff
@@ -0,0 +1,55 @@
+/*
+ * #224 Undead orc priest can die from fall damage
+ *
+ * Retrieve the symbol index of the NPC
+ */
+func int G1CP_224_OrcPriestFallDamage_GetInst() {
+    const int npcId = -2; // -1 is reserved for invalid symbols
+
+    if (npcId == -2) {
+        npcId = G1CP_GetNpcInstId("ORC_Priest_5");
+    };
+
+    return npcId;
+};
+
+/*
+ * This function applies the changes
+ */
+func int G1CP_224_OrcPriestFallDamage() {
+    var oCNpc npc; npc = Hlp_GetNpc(G1CP_224_OrcPriestFallDamage_GetInst());
+
+    if (!Hlp_IsValidNpc(npc)) {
+        return FALSE;
+    };
+    
+    if (npc.fallDownDamage != 10) {
+        return FALSE;
+    };    
+
+    npc.fallDownDamage = 0;
+    return TRUE;
```

Comment:
Although I understand the argument of avoiding too deeply nested if-blocks, I really feel that it is, on the one hand not necessary (it's just one nested if-block), and on the other hand creates three exit points of the function. Most importantly it creates much more code than necessary (and writing such a fix more work), and makes a concise and easily readable function much more complex and more difficult to follow at first glance.

The [previous flow of reading](https://g1cp.org/issues/224#implementation-draft) was "Set fallDownDamage to zero if all conditions are met", i.e. only do something if certain conditions are **true**. Now it is "If the Npc is ***in**valid*, return. If the fallDownDamage is ***not*** zero, return. Set the fallDownDamage to zero." The logical and semantic flow is lost and it takes a few moments to put these separated if-blocks together. By the multiple exit points of the functions it is harder to spot when there is actually something happening.

Semantically, I agree that "exiting early, when expected conditions are not met" makes a lot of sense for *test functions* (as they are currently mostly implemented). But for fix functions, it makes more sense the other way around: "fix something if all conditions are met". Only in the off-chance of stacking very, very many conditions or checking something general, e.g. if symbols exist, then it makes sense.

Reply:
The more important aspect for me is that the primary action is at the top level of the function, not "hidden" inside nested conditions. But I will most certainly think about this.